src/groupeddataframe/splitapplycombine.jl

```diff
@@ -950,12 +955,31 @@ function _combine_with_first!(first::Union{AbstractDataFrame,
     starts = gd.starts
     ends = gd.ends
     # Handle first group
-    j = append_rows!(first, outcols, colstart, colnames)
-    @assert j === nothing # eltype is guaranteed to match
-    append!(idx, Iterators.repeated(gdidx[starts[rowstart]], _nrow(first)))
+
+    @assert isempty(colnames) == (_ncol(first) == 0) == isempty(outcols)
+    if !isempty(colnames)
+        j = append_rows!(first, outcols, colstart, colnames)
+        @assert j === nothing # eltype is guaranteed to match
+        append!(idx, Iterators.repeated(gdidx[starts[rowstart]], _nrow(first)))
+    end
     # Handle remaining groups
     @inbounds for i in rowstart+1:len
         rows = wrap(do_call(f, gdidx, starts, ends, gd, incols, i))
+        if !(rows isa Union{AbstractDataFrame, NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}})
```

Comment:
This check was missing in the existing code?

Reply:
It would produce an error later (most likely inside `append_rows!`). I think it was better to have it here because:
1. it gives a cleaner error message early
2. in a `_ncol(rows)` test `rows` could be a `DataFrameRow` and if it had 0 columns we would allow it by `continue` and we should not.
3. a few lines below I do `if rows isa AbstractDataFrame` and in `else` I am sure it is a `NamedTuple` of vectors in `else` branch (otherwise I would have to add a third branch throwing an error).